src/motion-queue.h

```diff
@@ -105,9 +105,10 @@ class MotionQueue {
   // Shutdown. If !flush_queue: immediate, even if motors are still moving.
   virtual void Shutdown(bool flush_queue) = 0;
 
-  // Fill the argument with the current absolute position in loops
-  // for each motor.
-  virtual void GetMotorsLoops(MotorsRegister *absolute_pos_loops) = 0;
+  // Return the number of motion segments enqueued after
+  // the one currently executing and its already executed loops.
+  // buffer_size will never be 0 except if nothing has been yet enqueued.
+  virtual void Status(uint32_t *executed_loops, unsigned int *buffer_size) = 0;
```

Comment:
> The description is not entirely clear.
> You describe that 'returns the number of motion segments', but then the name of the variables does not reflect that. As reader, I could infer that it is buffer_size, but it is not entirely clear (later down in the code it become clear, that this is an ever increasing value until the queue has reached some equilibrum. I think this needs to be better explained in the method description or solved otherwise.
> (was this the question about sharing information about the queue size ?)

Yes precisely, 
the layer above motion-queue should be informed about the number of segments it has to keep hence not yet executed. 

So the when the queue beneath is full  (ie is executing the slot with index 5 and the last enqueued segment is the 4th) it returns 16, because I need to rollback of 16 segments. 

When instead all the segments are executed, motor-operations should keep just one segment so buffer_size is 1, and the number of loops returned will be 0 because the pru is waiting for a new slot to be enqueued.